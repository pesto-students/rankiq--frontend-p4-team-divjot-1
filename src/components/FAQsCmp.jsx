import { useTheme } from '@emotion/react';
import Paper from '@mui/material/Paper';
import Typography from '@mui/material/Typography';
import { useTranslation } from 'react-i18next';

function FAQsCmp() {
  const theme = useTheme();
  const { t } = useTranslation();
  const FAQentries = t('FAQs', { returnObjects: true });
  return (
    <Paper sx={{ paddingY: '50px' }}>
      {FAQentries.map((FAQentry) => (
        <>
          <Typography
            sx={{
              fontSize: '24px',
              fontWeight: 'light',
              paddingInline: '15%',
              paddingTop: '35px',
              [theme.breakpoints.down('md')]: {
                fontSize: '18px',
              },
            }}
          >
            {FAQentry.que}
          </Typography>
          <Typography
            sx={{
              fontSize: '16px',
              paddingInline: '15%',
              paddingTop: '5px',
              [theme.breakpoints.down('md')]: {
                fontSize: '12px',
              },
            }}
          >
            {FAQentry.ans}
          </Typography>
        </>
      ))}
    </Paper>
  );
}

export default FAQsCmp;
