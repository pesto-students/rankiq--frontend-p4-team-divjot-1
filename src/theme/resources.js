import enDefault from './l10n/en.json';
import hiDefault from './l10n/hi.json';

const resources = {
  en: {
    default: enDefault,
  },
  hi: {
    default: hiDefault,
  },
};

export default resources;
