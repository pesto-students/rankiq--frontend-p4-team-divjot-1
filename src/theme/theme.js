import { createTheme } from '@mui/material/styles';

const themeOptions = {
  palette: {
    type: 'light',
    primary: {
      main: '#3f51b5',
    },
    secondary: {
      main: '#ffffff',
    },
    text: { navFooter: '#FFFFFF' },
    box: { main: '#FAFAFA' },
  },
};

const theme = createTheme(themeOptions);

export default theme;
