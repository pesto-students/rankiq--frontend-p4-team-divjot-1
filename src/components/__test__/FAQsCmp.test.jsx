import { beforeEach, describe, expect, it, vi } from 'vitest';
import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import { ThemeProvider } from '@mui/material/styles';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import FAQsCmp from '../FAQsCmp';
import theme from '../../theme/theme';

const mockStore = configureStore([thunk]);
const navigate = vi.fn();
vi.mock('react-router-dom', async () => {
  return {
    ...(await vi.importActual('react-router-dom')),
    useNavigate: () => navigate,
  };
});

describe('FAQsCmp.jsx', () => {
  let tree;
  let store;
  const storeData = {
    auth: {
      userInfo: {},
      accessToken: 'testToken',
      loading: false,
    },
  };

  beforeEach(() => {
    store = mockStore(storeData);
    tree = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <FAQsCmp />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>
    );
  });

  it('should render without any errors', () => {
    expect(tree.asFragment()).toMatchSnapshot();
  });
});
