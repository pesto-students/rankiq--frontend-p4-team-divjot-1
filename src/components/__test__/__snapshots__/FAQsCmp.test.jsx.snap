// Vitest Snapshot v1

exports[`FAQsCmp.jsx > should render without any errors 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1qp5vf1-MuiPaper-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-eqynpf-MuiTypography-root"
      >
        Why my rank is changing each time I refresh the page?
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7zf3oe-MuiTypography-root"
      >
        The Rank iQ gives you a real-time analysis of your Rank which keeps getting updated as the number of students using this predictor keeps getting increased. So students are advised to check their ranks again after a certain time interval to know their updated rank.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eqynpf-MuiTypography-root"
      >
        What happens if the marks are same for multiple students? How rank is calculated?
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7zf3oe-MuiTypography-root"
      >
        Let’s say topper has 127 marks, so his rank is 1. Now, there are three students with 126 marks, then all of them will see their rank as 2. The student with 125 marks will see his rank as 1+3+1=5.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eqynpf-MuiTypography-root"
      >
        Do the total number of students increase after old student rechecking with the same URL?
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7zf3oe-MuiTypography-root"
      >
        Each student is given a unique Id to avoid any duplication. Hence, even if you check the marks with response sheet multiple times, total students don’t change.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eqynpf-MuiTypography-root"
      >
        The total number of student shown on rank page is for a particular branch or overall students?
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7zf3oe-MuiTypography-root"
      >
        The number of student shown on rank page is for particular branch or subjects.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-eqynpf-MuiTypography-root"
      >
        How Normalization score is calculated?
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7zf3oe-MuiTypography-root"
      >
        Normalization score is calculated in case of multiple shifts. Calculation varies as per exam conducting authority. Whichever is the exam, normalization score is calculated as per the formula given by exam conducting authority.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
